include/structopt/parser.hpp

```diff
@@ -169,25 +189,19 @@ struct parser {
   // Strip the initial dashes on the left of an optional argument
   // e.g., --verbose => verbose
   // e.g., -log-level => log-level
-  std::string lstrip_dashes(const std::string& next) {
-    std::string result;
-    bool prefix_dashes_ended = false;
-    for (auto& c : next) {
-      if (prefix_dashes_ended == false && c != '-') {
-        prefix_dashes_ended = true;
-      }
-      if (prefix_dashes_ended) {
-        result += c;
+  std::string_view lstrip_dashes(const std::string_view next) {
+    std::size_t n_dashes = 0;
+    while (n_dashes < next.size() && next[n_dashes] == '-') {
+      ++n_dashes;
       }
-    }
-    return result;
+    return trim_first_n(next, n_dashes);
   }
 
   // Get the optional field name if any from 
   // e.g., `-v` => `verbose`
   // e.g., `-log-level` => `log_level`
-  std::optional<std::string> get_full_optional_field_name(const std::string& next) {
-    std::optional<std::string> result;
+  std::optional<std::string_view> get_full_optional_field_name(const std::string_view next) {
```

Comment:
const argument without reference is useless. Make it const rvalue reference or remove const. (same problems at `visitor::operator()`)

Reply:
Personally I prefer always having consts on params, but I'm okay to take them off.